okhttp/src/test/java/okhttp3/ServerTruncatesRequestTest.kt

```diff
@@ -151,22 +161,72 @@ class ServerTruncatesRequestTest(
     makeSimpleCall()
   }
 
-  @Test fun serverTruncatesRequestButTrailersCanStillBeReadHttp1() {
+  @Test
+  fun serverDisconnectsBeforeSecondRequestHttp1() {
+    enableProtocol(Protocol.HTTP_1_1)
+
+    server.enqueue(MockResponse().setResponseCode(200).setBody("Req1"))
+    server.enqueue(MockResponse().setResponseCode(200).setBody("Req2"))
+
+    val eventListener = object : EventListener() {
+      var socket: SSLSocket? = null
+      var closed = false
+
+      override fun connectionAcquired(call: Call, connection: Connection) {
+          socket = connection.socket() as SSLSocket
+      }
+
+      override fun requestHeadersStart(call: Call) {
+        if (closed) {
+          throw IOException("fake socket failure")
```

Comment:
This isn't great but gets the failure to occur at the point we see it on Android.

```
  @Throws(IOException::class)
  fun writeRequestHeaders(request: Request) {
    try {
      eventListener.requestHeadersStart(call)
      codec.writeRequestHeaders(request)
      eventListener.requestHeadersEnd(call, request)
    } catch (e: IOException) {
      eventListener.requestFailed(call, e)
      trackFailure(e)
      throw e
    }
  }
```

Reply:
An alternative would be to create a DelegatingSocket or DelegatingSSLSocket and set a flag in requestHeadersStart, then cause the next write to fail.

Reply:
This is so nice and simple though.

Reply:
Thoughts on closing the socket here instead of throwing the exception? That way the exception comes from the right place in the code.